### ISSUE TITLE
Continuous recognition not re-starting after stopping

### ISSUE THREAD

**Issue by btaylor0** [Azure-Samples/cognitive-services-speech-sdk#881]
iOS Safari v14
Javascript 1.14 SDK

When using startContinuousRecognitionAsync Safari prompts correctly for microphone access, I have a 'microphone active' display next to the url, and speech gets recognized. If I call stopContinuousRecognitionAsync, and startContinuousRecognitionAsync on the same that same recognizer, the microphone appears to not be re-enabling. I do not get a microphone access prompt, and there is no 'microphone active' display next to the URL. If I call stopContinuousRecognitionAsync AGAIN the output is: 'NOMATCH: Speech could not be recognized'. This is happening if I have microphone permission set to either Ask, OR Allow.

**Comment by btaylor0** [comment]
I just noticed that this issue is not exclusive to iOS. It is happening in Chrome on Windows as well.

My current setup was working fine with SDK version 1.13, but I am now experiencing the above stated issues in multiple browsers and devices with v1.14

**Comment by glharper** [comment]
@btaylor0 Thanks for submitting this. Could you share the code you're using to reproduce this issue?

**Comment by btaylor0** [comment]
I call this function and pass it my button element, and text field.

It works on the first press, but if I press it to stop it, then press it again to start it, it appears to not be enabling the microphone. Again - this was working fine with v1.13, I just tested it again on the older version and it was good.

```
  function addSTTContinuousRecEvent(btnElement, textField) {
    let listening = false

    const continuousListening = (function() {
      const recognizer = new SpeechSDK.SpeechRecognizer(
        speechConfig,
        audioConfig,
      );

      recognizer.recognizing = (s, e) => {
        console.log(`RECOGNIZING: Text=${e.result.text}`);
      };

      recognizer.recognized = (s, e) => {
        if (e.result.reason == SpeechSDK.ResultReason.RecognizedSpeech) {
          textField.firstElementChild.value += `${e.result.text}`;
          console.log(`RECOGNIZED: Text=${e.result.text}`);
        } else if (e.result.reason == SpeechSDK.ResultReason.NoMatch) {
          console.log('NOMATCH: Speech could not be recognized.');
        }
      };

      recognizer.canceled = (s, e) => {
        console.log(`CANCELED: Reason=${e.reason}`);

        if (e.reason == CancellationReason.Error) {
          console.log(`"CANCELED: ErrorCode=${e.errorCode}`);
          console.log(`"CANCELED: ErrorDetails=${e.errorDetails}`);
          console.log('CANCELED: Did you update the subscription info?');
        }
      }
    
      function start() {
        recognizer.startContinuousRecognitionAsync();
      }
      function stop() {
        recognizer.stopContinuousRecognitionAsync();
      }
      function close() {
        recognizer.close();
      }
      return {
        start,
        stop,
        close
      }
    })()
    
    btnElement.addEventListener('click', event => {
      listening = !listening;
      if (listening) {
        continuousListening.start();
        btnElement.classList.add('stt_listening');
        btnElement.innerHTML = icons.microphoneFilled;
      } else {
        continuousListening.stop();
        btnElement.classList.remove('stt_listening');
        btnElement.innerHTML = icons.microphone;
      }
    });

```

**Comment by glharper** [comment]
@btaylor0 Thanks for sharing your code. One thing I see is that, in the case of a click registering twice quickly, the listening variable and the state of the recognizer will then be out of sync. Also, the recognizer is not in a listening state until the (currently null) callback is returned, as in:
```
      function start(cb, error_cb) {
        recognizer.startContinuousRecognitionAsync(cb, error_cb);
      }
[...]
      continuousListening.start( () => {
          btnElement.classList.add('stt_listening');
          btnElement.innerHTML = icons.microphoneFilled;
      }, 
      (err) => {
        console.log(err);
      });

```
But you are correct, this should mostly work. I have a version of this working correctly in Chrome on Windows using version 1.14.1 without the "listening" variable. What does your console output look like?

**Comment by btaylor0** [comment]
Thanks, I added the callbacks to the start/stop listening events. I also added console logs to show when it is listening/not listening.

![image](https://user-images.githubusercontent.com/50456198/101070428-5b07ad00-3569-11eb-9012-3d30300d5250.png)

![image](https://user-images.githubusercontent.com/50456198/101070458-63f87e80-3569-11eb-8d84-b16127e97635.png)

**Comment by glharper** [comment]
@btaylor0 Thanks again for opening this issue. I just checked in a fix which will be in the 1.15 release, due next month.
